src/NHibernate.Test/TypesTest/DateTimeTypeFixture.cs

```diff
@@ -167,12 +167,15 @@ public void UpdateUseExpectedSqlType()
 				t.Commit();
 			}
 
-			AssertSqlType(driver);
+			AssertSqlType(driver, 3);
 		}
 
 		[Test]
 		public void QueryUseExpectedSqlType()
 		{
+			if (!TestDialect.SupportsNonDataBoundCondition)
+				Assert.Ignore("Dialect does not support the test query");
```

Comment:
Debuging the failure, the expected DbType and SqlType usages count is found (5), but Firebird does not accept the last condition. Removing it allows the query to be executed, but I want this condition for avoiding having the Hql parser determining the type for all parameters from the properties they are compared to, and instead use the type supplied by the parameter. So I ignore this test for Firebird.